### PR TITLE
Enable CA1810 as a warning

### PR DESCRIPTION
**Pull request by pranavkm**
Contributes to https://github.com/dotnet/aspnetcore/issues/24055
Contributes to https://github.com/dotnet/aspnetcore/issues/32087

* This updates most of the repo to use LoggerMessageAttributes. MvcCoreLoggerExtensions is the only standout but it is fairly involved since it mixes logging from several types, re-uses logger ids etc

### CODE REVIEW

File: src/Mvc/Mvc.Razor/src/MvcRazorLoggerExtensions.cs [hunk 1]
```diff
@@ -1,204 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-using System.Diagnostics;
-using Microsoft.Extensions.Logging;
-
-namespace Microsoft.AspNetCore.Mvc.Razor;
```
Comment (pranavkm) [inline src/Mvc/Mvc.Razor/src/MvcRazorLoggerExtensions.cs:7]: Nearly all of the log messages on this types, except the one you see in the previous file, had been moved to individual types over time, and hence weren't referenced.

File: src/Mvc/Mvc.Core/src/MvcCoreLoggerExtensions.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 #nullable disable
+#pragma warning disable CA1810 // Initialize all static fields inline.
```
Comment (pranavkm) [inline src/Mvc/Mvc.Core/src/MvcCoreLoggerExtensions.cs:5]: I'll try and do do this separately. The source generator does not like that we reuse ids.

File: src/Localization/Localization/src/Internal/ResourceManagerStringLocalizerLoggerExtensions.cs [hunk 3]
```diff
@@ -1,26 +0,0 @@
...
-
-    static ResourceManagerStringLocalizerLoggerExtensions()
-    {
-        _searchedLocation = LoggerMessage.Define<string, string, CultureInfo>(
-            LogLevel.Debug,
-            new EventId(1, "SearchedLocation"),
-            $"{nameof(ResourceManagerStringLocalizer)} searched for '{{Key}}' in '{{LocationSearched}}' with culture '{{Culture}}'.");
-    }
-
-    public static void SearchedLocation(this ILogger logger, string key, string searchedLocation, CultureInfo culture)
```
Comment (pranavkm) [inline src/Localization/Localization/src/Internal/ResourceManagerStringLocalizerLoggerExtensions.cs:22]: Moved this to `ResourceManagerStringLocalizer` in the file below.

File: src/Analyzers/Analyzers/src/StartupAnalyzer.Diagnostics.cs [hunk 4]
```diff
@@ -53,5 +38,15 @@ static Diagnostics()
             DiagnosticSeverity.Warning,
             isEnabledByDefault: true,
             helpLinkUri: "https://aka.ms/AA64fv1");
+
+        public static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics = ImmutableArray.Create<DiagnosticDescriptor>(new[]
```
Comment (pranavkm) [inline src/Analyzers/Analyzers/src/StartupAnalyzer.Diagnostics.cs:42]: Static fields are initialized in the order in which they appear, but you can reference them before they are initialized. Cool new thing I found out today.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**